engine/script/src/script_sys.cpp

```diff
@@ -119,7 +119,7 @@ union SaveLoadBuffer
         FILE* file = fopen(tmp_filename, "wb");
         if (!file)
         {
-            return luaL_error(L, "Could not open the file %s.", tmp_filename);
+            return luaL_error(L, "Could not open the file %s, reason: %s.", tmp_filename, strerror(errno));
```

Comment:
strerror is not thread safe, but we use it http_cache.cpp. an alternative would be strerror_r for posix systems, but I don't know how portable it is?

Reply:
It seems like strerror_r has been around for quite some time:

https://man7.org/linux/man-pages/man3/strerror_l.3.html

Perhaps better to use it?

Reply:
The strerror_r is not standard, and might not be available on all platforms (MSVC, Switch, PS4/PS5?)

A newer function is called `strerror_s`, available from `C11`.
I'm not sure what the current supported set of features from our compilers is.
We'd have to test it.

Another option is to grab the source from somewhere, and put it in our code (license permitting).

Reply:
Can't use strerror_s on most platforms, but it seems I can use strerror_r in various forms depending on platform..